modules/crop/architecture.md

```diff
@@ -0,0 +1,50 @@
+# Architecture
+
+Diagram shows basic architecture of Crop API.
+
+![Crop API architecture](images/architecture.png)
+
+## Components
+
+Entity
+------
+- **Crop** Is a content entity that stores various information related to a crop area. It serves to store and returned all informations corresponding to crop area.
+- **CropType** It has a kind bunlde "CropType" who will inherit all the characteristics of the entity mother "Crop" but also allow a variation of Crop entity and will add the information management bind to a type of crop in particular.
+
+Plugin
+------
+- **CropEffect** Allows to apply a crop on an image via a ImageStyle effect.
+
+This diagram shows basic application that allows the effect "Manual Crop".
+
+![Manual Crop Effect](images/architecture_of_manual_crop.png)
+
+The effect "Manual Crop Effect" recover the crop has an image associated with two parameters.
+
+- **"uri"** This is an important propertie because it is the only contextual information available when you apply the effect "Manual Crop".
+- **"crop_type"** As it is possible to link CROP entity with same image, this is the only parameter that can distinguish several crop_type on the same image.
+
+Limitation :
+Currently Crop Api is not able to handle the case where one has several identical crop (uri & crop_type), in this case it returns the first entity found instead of list of corresponding entity.
```

Comment:
Due to limitations in Drupal core we're currently not not able to handle the case where one image would have several crops of the same crop type. The oldest crop will always be used in such case.